src/components/Button/Button.tsx

```diff
@@ -25,7 +25,9 @@ function Button(props: ButtonProps) {
 
 	const addAttrs: MosaicObject = {};
 
-	const MyButton = types[`${props.color}_${props.variant}`];
+	const textLink = props.variant === "text" && props.href ? "_link" : "";

```

Comment:
All of this is actually not needed, you already have everything you need on props and attrs. Also, when a prop.href get's passed, the Button component is smart enough to change the element from <button> to <a> and that means the only changes you have to do are on the css.